### ISSUE TITLE
ca1854 wrong suggestion when a value type is updated

### ISSUE THREAD

**Issue by luposky** [dotnet/roslyn-analyzers#6390]
### Analyzer

**Diagnostic ID**: [ca1854](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1854): `Prefer the IDictionary.TryGetValue(TKey, out TValue) method`

### Analyzer source

**SDK**: [Built-in CA analyzers in .NET 7 SDK or later](https://learn.microsoft.com/dotnet/fundamentals/productivity/code-analysis)

**Version**: [SDK 7.0.101](https://dotnet.microsoft.com/download/dotnet/7.0)

### Describe the bug

In my case the variable tagCount is a Dictionary<string, int>() and the suggested code refactoring produce unwanted and different result from the original code.

Original code:
```
if (tagCount.ContainsKey(tagId))
    tagCount[tagId]++;
else
    tagCount[tagId] = 1;
```

ca1854 suggested refactoring:
```
if (tagCount.TryGetValue(tagId, out var value))
    value++;
else
    tagCount[tagId] = 1;
```

### Steps To Reproduce
Run the program and inspect the values inside the Dictionary. The value++ statement does't update the dictionary, but a local copy of the value type variabile 'value'.

### Expected behavior
The values inside the Dictionary<string, int>() updated correctly as in the original source code.

### Actual behavior
Inside the Dictionary<string, int>() all values are 0.

**Comment by Youssef1313** [comment]
I'm not sure if CA1854 shouldn't report a diagnostic in this case, or if the code fix should be:

```csharp
if (tagCount.TryGetValue(tagId, out var value))
    tagCount[tagId] = value + 1;
else
    tagCount[tagId] = 1;
```

Best to benchmark, I think.

**Comment by Miizukii** [comment]
looks like TryGet is better, will add it to #6387 

|            Method |     Mean |    Error |   StdDev | Ratio |
|------------------ |---------:|---------:|---------:|------:|
|   TryGetIncrement | 21.83 ns | 0.132 ns | 0.124 ns |  0.68 |
| ContainsIncrement | 31.91 ns | 0.120 ns | 0.100 ns |  1.00 |

[Benchmark](https://gist.github.com/Miizukii/1917b448435ac20b6f87f6055f3d354d)
